contracts/Certificate.sol

```diff
@@ -360,11 +363,16 @@ contract Certificate is
    * @dev Reverts if the array lengths do not match.
    * @param removalIds Array of removal IDs.
    * @param removalAmounts Array of removal amounts.
+   * @param certificateAmount The total number of tonnes of carbon removals represented by the new certificate.
    */
   function _validateReceivedRemovalBatch(
     uint256[] calldata removalIds,
-    uint256[] calldata removalAmounts
+    uint256[] calldata removalAmounts,
+    uint256 certificateAmount
   ) internal pure {
+    if (removalAmounts.sum() != certificateAmount) {
+      revert("Incorrect supply allocation");
```

Comment:
note that I am intentionally choosing to use a traditional revert string because since this function is called from within `onERC1155BatchReceived` we have to rely on the try/catch error handling that's part of that implementation, and that error handling doesn't handle custom errors.  So if we use a custom error we fall through the try catch and revert with the far less correct and informative `'ERC1155: transfer to non ERC1155Receiver implementer'`